cmd/juju/application/deployer/bundle.go

```diff
@@ -186,13 +186,19 @@ func (d *deployBundle) checkExplicitSeries(bundleData *charm.BundleData) error {
 		// has the image-id constraint
 		machineHasImageID := false
 		for _, to := range applicationSpec.To {
-			machineCons, err := constraints.Parse(bundleData.Machines[to].Constraints)
-			if err != nil {
-				return errors.Trace(err)
-			}
-			if machineCons.HasImageID() {
-				machineHasImageID = true
-				break
+			// TODO(nvinuesa): here we only check for the image-id
+			// constraint when the placement directive is explicitly
+			// defined in the bundle. Find a way to check indirect
+			// placements like lxd:0 and ubuntu:0.
+			if machine, ok := bundleData.Machines[to]; ok {
```

Comment:
Although this prevents the panic, the logic to this isn't correct. You need to be able to resolve the machine, as that machine could have an ImageID.

Reply:
Yeah, I see. I have parsed the `to` directive, getting the machine from there. I still kept the `if` because machine can be `new` and in that case we need to avoid the check. See e5291868eff97a0711a1e5129d43f25058b1d824